src/check-nsfw.ts

```diff
@@ -1,192 +1,394 @@
-				inline: true
+				value: `<t:${Math.trunc(new Date().getTime() / 1000)}>`
 			},
 			{
-				name: predictions[1].className,
-				value: predictions[1].probability.toString(),
-				inline: true
+				name: 'Number of NSFW flags for this user (last 30 days)',
+				value: numberOfNsfwFlags.toString()
 			},
 			{
-				name: predictions[2].className,
-				value: predictions[2].probability.toString(),
-				inline: true
-			},
-			{
-				name: predictions[3].className,
-				value: predictions[3].probability.toString(),
-				inline: true
+				name: 'Thresholds (High / Low)',
+				value: `${messageClassifications.settings.highThreshold * 100}% / ${messageClassifications.settings.lowThreshold * 100}%`,
 			},
+			...topClassifiedImage.classifications.map(classification => {
+				return {
+					name: classification.className,
+					value: colorMessage(`${(classification.probability * 100).toFixed(2)}%`, classification.severity),
+					inline: true
+				};
+			}),
 			{
-				name: predictions[4].className,
-				value: predictions[4].probability.toString(),
-				inline: true
+				name: 'Hash',
+				value: `\`${binToHex(hash)}\``
 			}
 		]);
 
-		nsfwLogChannel.send({ embeds: [embed], files: suspectedFiles }).catch(err => {
-			nsfwLogChannel.send({ content: '`Unable to attach image`', embeds: [embed] });
-			console.log(err);
+	return [embed];
+}
+
+export async function handleAddNsfwExemption(interaction: ButtonInteraction): Promise<void> {
+	await interaction.deferReply();
+	let hash = interaction.message.embeds[0].fields
+		.find(field => field.name === 'Hash')!
+		.value;
+	hash = hexToBin(hash.substring(1, hash.length -1));
+	await NsfwExemption.create({ user_id: interaction.user.id, hash });
+	await interaction.editReply({ content: `Image added to NSFW exemptions by ${interaction.user.id}` });
+}
+
+export async function handleRemoveNsfwExemption(interaction: ButtonInteraction): Promise<void> {
+	await interaction.deferReply();
+	let hash = interaction.message.embeds[0].fields
+		.find(field => field.name === 'Hash')!
+		.value;
+	hash = hexToBin(hash.substring(1, hash.length -1));
+
+	const exemption = await NsfwExemption.findOne({ where: { hash } });
+	if (exemption) {
+		await exemption.destroy();
+		await interaction.editReply({ content: `Image removed from NSFW exemptions by ${interaction.user.id}` });
+	} else {
+		await interaction.editReply({ content: 'Image did not exist in NSFW exemptions' });
+	}
+}
+
+function colorMessage(message: string, type: ClassificationSeverity | null): string {
+	let color: string = '';
+	switch (type) {
+		case ClassificationSeverity.High:
+			color = '31';
+			break;
+		case ClassificationSeverity.Low:
+			color = '33';
+			break;
+	}
+
+	let prefix = '';
+	if (color !== '') {
+		prefix = `\u001b[0;${color}m`;
+	}
+
+	return `\`\`\`ansi\n${prefix}${message}\`\`\``;
+}
+
+function binToHex(string: string): string {
+	const chunks = string.match(/.{8}/g)?.map(chunk => parseInt(chunk, 2)) ?? [];
```

Comment:
This regex is not safe, consider the following binary string `111111111100000000`. This breaks into the following `11 11111111 00000000`. This regex however will create groups `[ '11111111', '11000000' ]`. The input string should be padded

Reply:
This isn't exported and will always be called safely, but I'm happy to update this